### PR TITLE
MELCloud integration (Mitsubishi Electric devices)

### PR DESCRIPTION
**Pull request by callemand**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.
- [x] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

The goal of this PR is to add communication between MELCloud and Gladys to be able to control air conditioning provide by Mitsubishi
For the moment only  air conditioning is available but we can add Air-To-Water and Energy-Recovery-Ventilation devices later.
Screenshot 2023-08-01 at 14 30 51

Screenshot 2023-08-01 at 14 31 19

Screenshot 2023-08-01 at 14 31 59
Screenshot 2023-08-01 at 14 31 36
Screenshot 2023-08-01 at 14 31 32

### CODE REVIEW

File: server/test/services/melcloud/lib/controllers/tuya.controller.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const sinon = require('sinon');
```
Comment (Pierre-Gilles) [inline server/test/services/melcloud/lib/controllers/tuya.controller.test.js]: tuya.controller?

File: server/services/melcloud/index.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+const MELCloudHandler = require('./lib');
+const { STATUS, MELCLOUD_ENDPOINT } = require('./lib/utils/melcloud.constants');
+
+module.exports = function MELCloudService(gladys, serviceId) {
+  const axios = require('axios');
+
+  const client = axios.create({
+    baseURL: MELCLOUD_ENDPOINT,
+    timeout: 1000,
```
Comment (Pierre-Gilles) [inline server/services/melcloud/index.js]: 1s timeout is a bit extreme no? on ADSL, could be easily reached. 5000 feels more safe :)

File: front/src/routes/integration/all/melcloud/device-page/DeviceTab.jsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    });
+    try {
+      const options = {
+        order_dir: this.state.orderDir || 'asc'
+      };
+      if (this.state.search && this.state.search.length) {
+        options.search = this.state.search;
+      }
+
+      const tuyaDevices = await this.props.httpClient.get('/api/v1/service/melcloud/device', options);
```
Comment (Pierre-Gilles) [inline front/src/routes/integration/all/melcloud/device-page/DeviceTab.jsx]: tuya?

File: front/src/routes/integration/all/melcloud/device-page/DeviceTab.jsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    this.getDevices();
+  }
+  async changeOrderDir(e) {
+    await this.setState({
+      orderDir: e.target.value
+    });
+    this.getDevices();
+  }
+
+  render({}, { orderDir, search, getTuyaStatus, tuyaDevices, housesWithRooms }) {
```
Comment (Pierre-Gilles) [inline front/src/routes/integration/all/melcloud/device-page/DeviceTab.jsx]: tuya

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
A few feedbacks! 

Be careful, there are a lot of references to the Tuya integration (variable names, file name, ..)

But otherwise, nothing serious, looks like a really good PR 😎

File: front/src/routes/integration/all/melcloud/setup-page/SetupTab.jsx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        '/api/v1/service/melcloud/variable/MELCLOUD_USERNAME'
+      );
+      melCloudUsername = username;
+
+      const { value: password } = await this.props.httpClient.get(
+        '/api/v1/service/melcloud/variable/MELCLOUD_PASSWORD'
+      );
+      melCloudPassword = password;
+      console.log('melCloudUsername', melCloudUsername);
+      console.log('melCloudPassword', melCloudPassword);
```
Comment (Pierre-Gilles) [inline front/src/routes/integration/all/melcloud/setup-page/SetupTab.jsx]: console.log

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Great PR 👏 
I have found just a tiny console.log remaining in the front, but nothing serious :) 

Otherwise, it's good for me!

File: front/src/routes/integration/all/melcloud/setup-page/SetupTab.jsx (hunk 5 above)
Comment (callemand) [reply front/src/routes/integration/all/melcloud/setup-page/SetupTab.jsx]: Fixed

**Pull request merged**